proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,147 @@
+# Opt-in `PrivateAssets` asset flow control option precedence over `IncludeAssets/ExcludeAssets` option
+
+- Author Name <https://github.com/erdembayar>
+- Start Date 2022-12-07
+- GitHub Issue <https://github.com/NuGet/Home/issues/6938>
+- GitHub PR <https://github.com/NuGet/NuGet.Client/pull/4976>
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+Currently `IncludeAssets/ExcludeAssets` options take precedence over `PrivateAssets` asset control option when creating package, so in many cases `IncludeAssets/ExcludeAssets` options completely eclipse the `PrivateAssets` option so doesn't let assets flow to consuming parent project. This proposal introduces new a `PackPrivateAssetsFlow` opt-in property so it let `PrivateAssets` asset control option takes precedence over `IncludeAssets/ExcludeAssets` options to enable asset flow to consuming parent projects.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+A package author should be able control which asset flow to parent consuming projects when this project is consumed as package. It enables parent projects to consume transitive packages without having to directly reference them, therefore it reduces the number of packages developers need to keep track of.
+We couldn't make this default experience because it'll break customers who rely on current behavior.
+
+## Explanation
```

Comment:
There needs to be a section about how `dotnet add package` and Installing packages via Visual Studio (PMC's `Install-Package` or PM UI) will add `PrivateAssets="all"` when a package's nuspec declares it a development dependency.

If this proposal goes ahead as planned (project level property to change `PrivateAssets`'s meaning to "assets to flow transitively" instead of the existing "assets to not flow transitively"), then whoever implements this fearture needs to additionally change both `dotnet add package` and VS package install, to detect when this project level property is enabled and change the include/private/exclude assets to keep the intent of "development package" (package will not be a dependency if proejct is packed).  Otherwise, this is a breaking change.

Reply:
> There needs to be a section about how dotnet add package and Installing packages...

That seems irrelevant to this feature. I don't know why we'd document it here.

> project level property to change PrivateAssets's meaning to "assets to flow transitively" instead of the existing "assets to not flow transitively"

That is not at all what this feature is about, which I elaborate on in a reply to one of your other comments.

Reply:
I think I'm starting to get it.  The opt-in feature would make `PrivateAssets` be the exclusion list for the dependency when the package is packed, regardless of `IncludeAssets` and `ExcludeAssets`, right?

Reply:
Right. Also (hopefully) it will impact parent projects (P2Ps that reference this project) -- not just folks consuming it via PackageReference.

Reply:
> Right. Also (hopefully) it will impact parent projects (P2Ps that reference this project) -- not just folks consuming it via PackageReference.

Agreed. In fact, I'd go as far to say that it has to work that way.

Reply:
Resolving this thread considering we have enough consensus on what problem we're solving.